### PR TITLE
General overhaul

### PR DESCRIPTION
**Pull request by teutoburg**
- Migrate to `pyproject.toml` using Poetry
- Remove redundant or empty files
- Improve README a little
- Use DevOps-based test workflow
- Include recent workflows for bump, publish and release
- Increase some version requirements and fix some tests accordingly

Interestingly, most of the sky background tests now fail. This might have to do with ESO's skycalc server move to 002. Needs further investigation, for now skipping those tests.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -3,32 +3,15 @@ name: Tests
...
   # Allows you to run this workflow manually from the Actions tab
   workflow_dispatch:
 
 jobs:
-  build:
-    runs-on: ${{ matrix.os }}
-    strategy:
-      matrix:
-        os: [ubuntu-latest]
-        python-version: ['3.7', '3.8', '3.9']
```
Comment (hugobuddel) [inline .github/workflows/tests.yml:20]: Waaaaaaaaaaaaaaa

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+[tool.poetry]
+name = "HowManyPhotons"
+version = "0.3.0a0"
+description = "Simple photon count utility for astronomical fluxes"
+authors = ["Kieran Leschinski <kieran.leschinski@unive.ac.at>"]
+maintainers = [
+    "Kieran Leschinski <kieran.leschinski@unive.ac.at>",
+    "Hugo Buddelmeijer <hugo@buddelmeijer.nl>",
+    "Fabian Haberhauer <fabian.haberhauer@univie.ac.at>",
```
Comment (hugobuddel) [inline pyproject.toml:9]:
```suggestion
    "Fabian Haberhauer <fabian.haberhauer@univie.ac.at>",
    "Hugo Buddelmeijer <hugo@buddelmeijer.nl>",
```

File: hmbp/main.py [hunk 3]
```diff
@@ -133,7 +133,7 @@ def for_flux_in_filter(filter_name, flux, instrument=None, observatory=None,
             flux = flux.value
         spec = spec_template(flux)
 
-    wave = filt.waveset if len(filt.waveset) > len(spec.waveset) else spec.waveset
+    wave = max(filt.waveset, spec.waveset, key=len)
```
Comment (hugobuddel) [inline hmbp/main.py:136]: Nice

**Review by hugobuddel — APPROVED** [review]
Hell yeah! Thanks, I had not looked at HowManyBloodyPhotons at all.

We should understand why the tests fail though... Some tests are like 6 orders of magnitude off now (on `main`), which is indeed suspicious with the skycalc mm -> um change.

File: .github/workflows/tests.yml (hunk 1 above)
Comment (teutoburg) [reply .github/workflows/tests.yml:20]: indeed 😄

File: pyproject.toml (hunk 2 above)
Comment (teutoburg) [reply pyproject.toml]: If you insist 😉

**Pull request merged**